src/device/rcp/vi/vi_controller.c

```diff
@@ -157,6 +158,9 @@ void write_vi_regs(void* opaque, uint32_t address, uint32_t value, uint32_t mask
 void vi_vertical_interrupt_event(void* opaque)
 {
     struct vi_controller* vi = (struct vi_controller*)opaque;
+
+    netplay_check_sync(&vi->mi->r4300->cp0);
```

Comment:
This code should go inside new_vi() function so that netplay code doesn't mix with n64 emulation code.

Reply:
The reason I put it here is because it needs to read the CP0 register values, if I did it in new_vi, I would need to read the cp0 values using the global g_dev variable I would think, which method is preferable?

Reply:
I'd say using g_dev inside new_vi is better here because of better separation of concerns netplay vs emulation.